core/src/main/java/com/linecorp/armeria/common/ContextAwareEventLoop.java

```diff
@@ -32,18 +36,51 @@
 import io.netty.util.concurrent.ScheduledFuture;
 
 /**
- * A delegating {@link EventLoop} that makes sure all submitted tasks are
- * executed within the {@link RequestContext}.
+ * A delegating {@link EventLoop} that sets the {@link RequestContext} before executing any submitted tasks.
  */
-final class RequestContextAwareEventLoop extends RequestContextAwareExecutorService implements EventLoop {
+public final class ContextAwareEventLoop extends ContextAwareExecutorService implements EventLoop {
+
+    /**
+     * Returns a new {@link ContextAwareEventLoop} that sets the specified {@link RequestContext}
+     * before executing any submitted tasks.
+     */
+    public static ContextAwareEventLoop of(RequestContext context, EventLoop eventLoop) {
+        requireNonNull(context, "context");
+        requireNonNull(eventLoop, "eventLoop");
+        if (eventLoop instanceof ContextAwareEventLoop) {
+            final RequestContext ctx = ((ContextAwareEventLoop) eventLoop).context();
+            if (context == ctx) {
+                return (ContextAwareEventLoop) eventLoop;
+            }
+            throw new IllegalArgumentException(
+                    "cannot create a " + ContextAwareEventLoop.class.getSimpleName() +
+                    " using another " + eventLoop);
+        }
+        return new ContextAwareEventLoop(context, eventLoop);
+    }
+
+    private final EventLoop eventLoop;
 
-    RequestContextAwareEventLoop(RequestContext context, EventLoop delegate) {
-        super(context, delegate);
+    private ContextAwareEventLoop(RequestContext context, EventLoop eventLoop) {
+        super(context, eventLoop);
+        this.eventLoop = eventLoop;
     }
 
+    /**
+     * Returns the {@link EventLoop} that is executing submitted tasks without setting
+     * the {@link RequestContext}.
+     */
+    public EventLoop detachContext() {
```

Comment:
This method name seems mutable, not sure it follows our normal conventions.

`withoutContext`?

Reply:
That's a good suggestion. Fixed. 😉

Reply:
`withoutContext` looks good to me. On second thought, we can implement `Unwrappable` and seems to use `unwrap`. (You can ignore this review.😉)

Reply:
Let's use `withoutContext` as we've discussed. 😉